scripts/generate_conda_yaml.py

```diff
@@ -0,0 +1,73 @@
+from __future__ import annotations
+
+import argparse
+import pathlib
+import sys
+
+import toml
+import yaml
+
+PROJECT_DATA = toml.load(
+    pathlib.Path(__file__).parent.parent / 'pyproject.toml'
+)['project']
+
+PROJECT_MAP = {
+    'graphviz': 'python-graphviz'
+}
+
+def normalize_dependency(dep):
+    for proj, conda_dep in PROJECT_MAP.items():
+        if proj in dep:
+            dep = dep.replace(proj, conda_dep)
+    return dep.replace('==', '=').replace(' =', '=').replace(' >', '>').replace(' <', '<')
+
+def generate_yaml(
+    name: str = None,
+    python_version: str = '3.10',
+    optional: List[str] | None = None,
+    channels: List[str] | None = None,
+    extras:   List[str] | None = None,
+    pip_only: List[str] | None = None
+):
+    spec = {}
+    if name:
+        spec['name'] = name
+    if channels:
+        spec['channels'] = channels
+    spec['dependencies'] = deps = [f'python={python_version}'] if python_version else []
+    deps += ['pip']
+    deps += [
+        normalize_dependency(dep) for dep in PROJECT_DATA['dependencies']
+    ]
+    optional_deps = PROJECT_DATA['optional-dependencies']
+    pip_deps = [dep for p in pip_only for dep in optional_deps[p]]
+    for opt in (optional or []):
```

Comment:
We should support something like this here: `datashader >=0.11.1; python_version < '3.11'`. I have some janky code I wrote for this when testing Nox:

``` python
from packaging.markers import Marker

def _install_mapping(*pypi_packages, python_version=None):
    mapping = {"ibis-framework": "ibis-sqlite"}
    env = {"python_version": python_version}

    conda_packages = []
    for package in pypi_packages:
        for pypi_name, conda_name in mapping.items():
            if package.startswith(pypi_name):
                package.replace(pypi_name, conda_name)
                break
        if ";" in package and python_version:
            package, *markers = package.split(";")
            if not all(Marker(m).evaluate(environment=env) for m in markers):
                continue

        conda_packages.append(package)
    return conda_packages
```